classwork/controlWork1/priorityQueue/priorityQueue/PriorityQueueParallel.cs

```diff
@@ -0,0 +1,71 @@
+﻿public class PriorityQueueParallel<TPriority, TValue> where TPriority : IComparable
+{
+    private readonly object _lockObject = new();
+    private readonly SortedDictionary<TPriority, Queue<TValue>> _buffer;
+    private int _size;
+
+    /// <summary>
+    /// Getting size of the queue.
+    /// </summary>
+    public int Size
+    {
+        get
+        {
+            lock (_lockObject)
+            {
+                return _size;
+            }
+        }
+    }
+
+    public PriorityQueueParallel()
+    {
+        _buffer = new SortedDictionary<TPriority, Queue<TValue>>();
+    }
+
+    /// <summary>
+    /// Dequeuing value of element with highest priority. If empty, waiting for enqueuing and then dequeuing.
+    /// </summary>
+    /// <returns>Value of element with the highest priority.</returns>
+    public TValue Dequeue()
+    {
+        lock (_lockObject)
+        {
+            if (_size == 0)
```

Comment:
```suggestion
            while (_size == 0)
```
А если несколько потоков заблокируются на `Wait`? Затем в очередь добавят элемент, все эти потоки проснутся. Один из них возьмет элемент из очереди, а остальные упадут на `queue.Keys.Last()`. 
Я ведь говорил на паре, что не `if`, а `while` тут нужно использовать.

Reply:
-1